caluma_interval/interval.py

```diff
@@ -16,6 +17,15 @@
 __author__ = "Adfinis SyGroup"
 
 
+ISO8601_PERIOD_DATE_REGEX = re.compile(
+    r"^P(?!$)"
+    r"(\d+(?:[,.]\d+)?Y)?"
+    r"(\d+(?:[,.]\d+)?M)?"
+    r"(\d+(?:[,.]\d+)?W)?"
+    r"(\d+(?:[,.]\d+)?D)?$"
+)
```

Comment:
Hint: If you're using the `re.X` flag (or `re.VERBOSE`), you can write regexes with spaces and newlines in them that aren't matched, which allows you to write longer regexes in a more readable way:
```python
 ISO8601_PERIOD_DATE_REGEX = re.compile(r""" 
     ^P(?!$)          # explain this line 
     (\d+(?:[,.]\d+)?Y)?   # explain that line
     (\d+(?:[,.]\d+)?M)? 
     (\d+(?:[,.]\d+)?W)? # ... and so on
     (\d+(?:[,.]\d+)?D)? 
     $ 
 """, re.VERBOSE)                                                        
```